odoo/addons/base/models/ir_attachment.py

```diff
@@ -650,12 +650,32 @@ def generate_access_token(self):
     def _generate_access_token(self):
         return str(uuid.uuid4())
 
+    def validate_access(self, access_token):
+        self.ensure_one()
+        record_sudo = self.sudo()
```

Comment:
```suggestion
        record_sudo = self.sudo().with_context(prefetch_fields=False)
```
And you can remove your crap about _cache.clear in _find_record.